### PR TITLE
PVP 109 added task cards fore each type of task

### PR DESCRIPTION
**Pull request by vytjar**
Dummy tasks for preview created at TaskScreen.

### CODE REVIEW

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 1]
```diff
@@ -1,18 +1,222 @@
...
+
+    return buildString {
+        if (hours > 0) append("$hours h")
+        if (minutes > 0) append("${if (hours > 0) " " else ""}$minutes min")
+    }
+}
+@Composable
+fun MealTaskBoxBody(
+    mealTask: MealTask
+){
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:50]: Single spacing before opening curly brackets should be added. `Ctrl+Alt+L` hotkey should format it automatically.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 2]
```diff
@@ -1,18 +1,222 @@
...
+
+    Text(
+        "Main ingredient: " + mealTask.recipe,
+        textAlign = TextAlign.Left,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 8.dp)
+    )
+
+    if(mealTask.duration != null){
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:60]: Same as above regarding the space. Remains throughout the whole file.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 3]
```diff
@@ -1,18 +1,222 @@
...
+        Text(
+            "Duration: " + getDurationString(mealTask.duration!!),
+            textAlign = TextAlign.Left,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 8.dp)
+        )
+    }
+}
+@Composable
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:70]: Should add an empty single line to separate components and increase readability.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 4]
```diff
@@ -1,18 +1,222 @@
...
+fun SportTaskBoxBody(
+    sportTask: SportTask
+) {
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:74]: Redundant empty line.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 5]
```diff
@@ -1,18 +1,222 @@
...
+            textAlign = TextAlign.Left,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 8.dp)
+        )
+    }
+}
+@Composable
+fun SportTaskBoxBody(
+    sportTask: SportTask
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:72]: Could be named `task`, there are no other type tasks inside anymore and type is known.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 6]
```diff
@@ -1,18 +1,222 @@
...
+) {
+
+    sportTask.activity?.let { activity ->
+        val activityText = buildString {
+            append(activity.title)
+            when {
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:78]: Two separate actions: `method call` and `when logic body`. Should be separated by a single empty line to increase readability.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 7]
```diff
@@ -1,18 +1,222 @@
...
+                .padding(start = 8.dp)
+        )
+    }
+}
 
+@Composable
+fun TaskBoxBody(
+    task: Task
+) {
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:101]: Redundant empty line.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 8]
```diff
@@ -1,18 +1,222 @@
...
+        Text(
+            text = "Duration: ${getDurationString(duration)}",
+            textAlign = TextAlign.Left,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 8.dp)
+        )
+    }
+}
 @Composable
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:122]: A single empty line should be added.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 9]
```diff
@@ -1,18 +1,222 @@
...
 fun TaskBox(
     task: Task
 ) {
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:126]: Redundant empty line.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 10]
```diff
@@ -1,18 +1,222 @@
...
+        mutableStateOf(false)
+    }
+
+    Card(
+        shape = RectangleShape,
+        modifier = Modifier
+            .fillMaxWidth()
+            .border(BorderStroke(1.dp, MaterialTheme.colorScheme.outline))
+    ) {
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:138]: Redundant empty line.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 11]
```diff
@@ -1,18 +1,222 @@
...
+        Column(
+        ) {
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:141]: Redundant empty line.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 12]
```diff
@@ -1,18 +1,222 @@
...
+    Card(
+        shape = RectangleShape,
+        modifier = Modifier
+            .fillMaxWidth()
+            .border(BorderStroke(1.dp, MaterialTheme.colorScheme.outline))
+    ) {
+
+        Column(
+        ) {
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:140]: Empty brackets should stay in single line.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 13]
```diff
@@ -1,18 +1,222 @@
...
+
+            Row(
+                modifier = Modifier.fillMaxWidth(),
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:143]: Single parameter within brackets _could_ stay in a single line. Can leave it as it is, if wanted.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 14]
```diff
@@ -1,18 +1,222 @@
...
+            ) {
+                Checkbox(
+                    checked = checked,
+                    onCheckedChange = {checked = it},
+                    modifier = Modifier
+                        .size(36.dp)
+                        .align(CenterVertically)
+                )
+                Text(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:152]: A single empty line should be added.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 15]
```diff
@@ -1,18 +1,222 @@
...
+            }
+
+            val timeString = "Scheduled at ${task.scheduledAt
+                .toLocalTime()
+                .format(DateTimeFormatter.ofPattern("HH:mm"))}"
+
+            Text(
+                timeString,
+                modifier = Modifier
+                    .padding(start = 8.dp)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:177]: Single method call should stay within single line.

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 16]
```diff
@@ -1,18 +1,222 @@
...
+    val task = Task(
+        description = "Complete project tasks",
+        duration = Duration.ofHours(2),
+        id = "3",
+        isCompleted = false,
+        scheduledAt = LocalDateTime.now(),
+        title = "Project Tasks",
+        userEmail = "example@example.com"
+    )
+    TaskBox(sportTask)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:221]: Single empty line should be added to separate different elements (variable, component).
Also, a single empty line (line-break) could be added to the end of the file to support UNIX file endings correctly as everywhere else.

**Review by klaidoshka — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt [hunk 17]
```diff
@@ -1,18 +1,222 @@
...
+    // Later task.isCompleted should be used
+    var checked by remember {
+        mutableStateOf(false)
+    }
+
+    Card(
+        shape = RectangleShape,
+        modifier = Modifier
+            .fillMaxWidth()
+            .border(BorderStroke(1.dp, MaterialTheme.colorScheme.outline))
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt:136]: Multiple parameters should be split into multiple lines.

**Comment by klaidoshka** [comment]
Lines :211 and :125 _could_ be further modified, but either way, it is way better than before 😊

Also, `#getDurationString` method documentation has triple stars for comment `/***`, I would guess that it was done manually. You can set it automatically by writing `/**` and pressing `ENTER`.

For Java it would also fill parameters, their names, return type, exception type (if it is annotated).
```java
/**
 * <p>This is a simple description of the method. . .
 * <a href="http://www.x.com">X!</a>
 * </p>
 * @param incomingDamage the amount of incoming damage      <-- auto generated, except explanation
 * @return the amount of health hero has after attack       <-- auto generated, except explanation
 * @see <a href="http://www.link_to_jira/HERO-402">HERO-402</a>
 * @since 1.0
 */
 public static int damageHero(int incomingDamage) {
     // ...
 }
 ```
However, for kotlin it is not the case, it seems. Kotlin decided to make documentation comments more free-ish by not setting any template. How to use it to greater extent for better developing experience can be found [here](https://kotlinlang.org/docs/kotlin-doc.html#kdoc-syntax).

**Review by klaidoshka — APPROVED** [review]

**Review by KhajiitLikeToSneak — APPROVED** [review]

**Pull request merged**